src/library_syscall.js

```diff
@@ -280,7 +280,9 @@ var SyscallsLibrary = {
     if (len === info.len) {
 #if FILESYSTEM && SYSCALLS_REQUIRE_FILESYSTEM
       var stream = FS.getStream(info.fd);
-      SYSCALLS.doMsync(addr, stream, len, info.flags, info.offset);
+      if (info.prot & 0x02) {
```

Comment:
I believe the constant can be `{{{ cDefine('PROT_WRITE') }}}`.

Reply:
thanks for reviewing @kripken . Replaced with the constant now.